src/Server/TextDocument.php

```diff
@@ -124,4 +127,25 @@ private function updateAst(string $uri, string $content)
             $this->asts[$uri] = $stmts;
         }
     }
+
+    /**
+     * The document formatting request is sent from the server to the client to format a whole document.
+     *
+     * @param TextDocumentIdentifier $textDocument The document to format
+     * @param FormattingOptions $options The format options
+     * @return TextEdit[]
+     */
+    public function formatting(TextDocumentIdentifier $textDocument, FormattingOptions $options)
+    {
+        $nodes = $this->asts[$textDocument->uri];
+        if (empty($nodes)) {
+            return [];
+        }
+        $prettyPrinter = new PrettyPrinter();
+        $edit = new TextEdit();
+        $edit->range = new Range(new Position(0, 0), new Position(PHP_INT_MAX, PHP_INT_MAX));
```

Comment:
@dbaeumer Of course a minimal text edit is better, but the default implementation of PHPParser's PrettyPrinter will just spit out the new whole new content. We may customize the behavior later, but for now we just wanted to get the feature done. My question was only about wether it is okay to use a very big number (PHP_INT_MAX) to make it update the whole document (even though the EOF offset is much smaller) or if that is not protocol-conform and causes problems.